versions/development/SPEC.md

```diff
@@ -803,20 +808,27 @@ task t {
   input {
     Int i
     File f
+    Directory d
   }
 
   # [... other task sections]
 }
 ```
 
 #### Task Input Localization
-`File` inputs must be treated specially since they require localization to within the execution directory:
-- Files are localized into the execution directory prior to the task execution commencing.
+`File` and `Directory` inputs must be treated specially since they require localization to within the execution directory:
+- Files and directories are localized into the execution directory prior to the task execution commencing. 
 - When localizing a `File`, the engine may choose to place the file wherever it likes so long as it accords to these rules:
   - The original file name must be preserved even if the path to it has changed.
   - Two input files with the same name must be located separately, to avoid name collision.
   - Two input files which originated in the same storage directory must also be localized into the same directory for task execution (see the special case handling for Versioning Filesystems below).
+- When localizing a `Directory`, the engine follows the same set of rules as for Files:
+  - The original directory name must be preserved even if the path to it has changed.
+  - Two input directories with the same name must be located separately, to avoid name collision.
+  - Two input directories which originated in the same parent storage directory must be localized into the same parent directory for task execution.
```

Comment:
What does this mean? Does it mean if you have input directory `/a/b/c` and input directory `/a/b/d` then the localized paths must look like `"/some/prefix/b/c"` and `/some/prefix/b/d`? If so, does this only apply to a directory's immediate parent? Some examples might help to make required behavior clear.

Reply:
Just a reminder that there'll be no changes as part of this PR at this stage. If the answers from @cjllanwarne to your questions lead you towards a yes or no vote, then great.

It's also AOK to suggest followup changes (e.g. more examples)

Reply:
I would like to see a few more examples please -- particularly toward documenting behavior wrt the point raised by @pshapiro4broad

Reply:
1. As written, I think that this line is a pretty weak guarantee: that the directories would end up in `/some/prefix/c` and `/some/prefix/d`. Of course an engine is free to include more of the original path within `/some/prefix/` if it chooses to (as Cromwell does in the prototype https://github.com/broadinstitute/cromwell/pull/3980).
1. This was added to mirror the guarantees that exist for files (eg so that `.bam` and `.bai` inputs remain co-located if they start off next to each other).
1. I agree that it is a little ambiguous as-is, so (assuming this PR merges) I can make a clarification/alteration/examples PR to follow with.

Reply:
(or, if you think this is gating, you can vote against this as-is and request I resubmit with that change made)

Reply:
Thanks for the clarification. It doesn't seem worth rejecting based on this, but making the clarification part of a future change to the spec would be nice.